api-server/src/http/controllers/food.controller.ts

```diff
@@ -1,49 +1,80 @@
 import { Request, Response } from 'express';
-import { FoodLocal, PortionSizeMethod, PortionSizeMethodParameter } from '@/db/models/foods';
-import { NotFoundError } from '../errors';
+import { IoC } from '@/ioc';
+import { FoodDataEntryResponse } from '@common/types/http';
 import { Controller } from './controller';
 
-type FoodData = {
-  [key: string]: any; // FIXME: use more specific type
-};
-
 export type FoodController = Controller<
   'entry' | 'entryWithSource' | 'brands' | 'associatedFoods' | 'composition'
 >;
 
-export default (): FoodController => {
+export default ({ foodDataService }: IoC): FoodController => {
```

Comment:
I would type it: `Pick<IoC, 'foodDataService'>` to allow it testable if needed.

Reply:
yep, thanks for the suggestion.